subsys/net/l2/openthread/Kconfig

```diff
@@ -179,45 +180,51 @@ config OPENTHREAD_MBEDTLS_LIB_NAME
 	  This option allows to specify one or more mbedtls library files to be
 	  linked with OpenThread. Separate multiple values with space " ".
 
-menuconfig OPENTHREAD_NCP
-	bool "Network Co-Processor"
+config OPENTHREAD_COPROCESSOR
+	bool "OpenThread Co-Processor"
 	select OPENTHREAD_MANUAL_START
 	select RING_BUFFER
 	select UART_INTERRUPT_DRIVEN
 	help
-	  Enable NCP in OpenThread stack.
+	  Enable Co-Processor in OpenThread stack.
 
-if OPENTHREAD_NCP
-
-config OPENTHREAD_NCP_RADIO
-	bool "Network Co-Processor as Radio"
+choice
+	prompt "OpenThread Co-Processor type"
 	help
-	  Enable NCP in OpenThread stack as radio-only.
+	  This option selects Thread network co-processor type
+
+config OPENTHREAD_COPROCESSOR_NCP
+	bool "NCP - Network Co-Processor"
+config OPENTHREAD_COPROCESSOR_RCP
+	bool "RCP - Radio Co-Processor"
+endchoice
+
+if OPENTHREAD_COPROCESSOR
 
-config OPENTHREAD_NCP_SPINEL_ON_UART_DEV_NAME
-	string "UART device to use for NCP SPINEL"
+config OPENTHREAD_COPROCESSOR_SPINEL_ON_UART_DEV_NAME
+	string "UART device to use for Co-Processor SPINEL"
 	default "UART_0"
 	help
-	  UART device to use for NCP SPINEL.
+	  UART device to use for Co-Processor SPINEL.
 
-config OPENTHREAD_NCP_UART_RING_BUFFER_SIZE
-	int "Set NCP UART ring buffer size"
+config OPENTHREAD_COPROCESSOR_UART_RING_BUFFER_SIZE
+	int "Set Co-Processor UART ring buffer size"
 	default 4096
 	help
-	  TX buffer size for the OpenThread NCP UART.
+	  TX buffer size for the OpenThread Co-Processor UART.
 
-config OPENTHREAD_NCP_SPINEL_ON_UART_ACM
+config OPENTHREAD_COPROCESSOR_SPINEL_ON_UART_ACM
 	bool "Run SPINEL over USB-ACM"
 	depends on USB_CDC_ACM
 	help
 	  Is the SPINEL device a USB-CDC-ACM device.
 
-config OPENTHREAD_NCP_VENDOR_HOOK_SOURCE
-	string "Path to NCP vendor hook source file"
+config OPENTHREAD_COPROCESSOR_VENDOR_HOOK_SOURCE
```

Comment:
@kkasperczyk-no are vendor hooks NCP specific, or are they applicable for RCP as well? If they're for NCP only, I'd leave the config name as is, but add `OPENTHREAD_COPROCESSOR_NCP` dependency.

Reply:
As far as I know it could be used in RCP as well to extend Spinel commands list and manage radio driver functionalities. @MarekPorwisz could you confirm my words?

Reply:
That's my assumption as well, but I have not tested Vendor hooks on RCP, yet.